examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -0,0 +1,885 @@
+      "        index into arrays. A typical use case is when arrays are accessed\n",
+      "        indirectly via the ``condition`` expression. Defaults to False.\n",
+      "\n",
+      "    Examples\n",
+      "    --------\n",
+      "    Among the other things, ConditionalDimensions are indicated to implement\n",
+      "    Function subsampling. In the following example, an Operator evaluates the\n",
+      "    Function ``g`` and saves its content into ``f`` every ``factor=4`` iterations.\n",
+      "\n",
+      "    >>> from devito import Dimension, ConditionalDimension, Function, Eq, Operator\n",
+      "    >>> size, factor = 16, 4\n",
+      "    >>> i = Dimension(name='i')\n",
+      "    >>> ci = ConditionalDimension(name='ci', parent=i, factor=factor)\n",
+      "    >>> g = Function(name='g', shape=(size,), dimensions=(i,))\n",
+      "    >>> f = Function(name='f', shape=(int(size/factor),), dimensions=(ci,))\n",
+      "    >>> op = Operator([Eq(g, 1), Eq(f, g)])\n",
+      "\n",
+      "    The Operator generates the following for-loop (pseudocode)\n",
+      "\n",
+      "    .. code-block:: C\n",
+      "\n",
+      "        for (int i = i_m; i <= i_M; i += 1) {\n",
+      "          g[i] = 1;\n",
+      "          if (i%4 == 0) {\n",
+      "            f[i / 4] = g[i];\n",
+      "          }\n",
+      "        }\n",
+      "\n",
+      "    Another typical use case is when one needs to constrain the execution of\n",
+      "    loop iterations to make sure certain conditions are honoured. The following\n",
+      "    artificial example employs indirect array accesses and uses ConditionalDimension\n",
+      "    to guard against out-of-bounds accesses.\n",
+      "\n",
+      "    >>> from sympy import And\n",
+      "    >>> ci = ConditionalDimension(name='ci', parent=i,\n",
+      "    ...                           condition=And(g[i] > 0, g[i] < 4, evaluate=False))\n",
+      "    >>> f = Function(name='f', shape=(int(size/factor),), dimensions=(ci,))\n",
+      "    >>> op = Operator(Eq(f[g[i]], f[g[i]] + 1))\n",
+      "\n",
+      "    The Operator generates the following for-loop (pseudocode)\n",
+      "\n",
+      "    .. code-block:: C\n",
+      "\n",
+      "        for (int i = i_m; i <= i_M; i += 1) {\n",
+      "          if (g[i] > 0 && g[i] < 4) {\n",
+      "            f[g[i]] = f[g[i]] + 1;\n",
+      "          }\n",
+      "        }\n",
+      "\n",
+      "    \n"
+     ]
+    }
+   ],
+   "source": [
+    "from devito import ConditionalDimension\n",
+    "print(ConditionalDimension.__doc__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use case A (honour a certain condition)\n",
+    "\n",
+    "\n",
+    "## Defining a condition\n",
+    "Now it's time to show a more descriptive example. Let's define a conditional that updates the values of the `Function` that are larger than zero, f(x,y) > 0. We name our `ConditionalDimension` `ci`.\n",
```

Comment:
Drop "Now" at the beginning.

"Let's define a conditional that updates the values" -> "We define a conditional that increments by one all values of a `Function` that are larger than zero." -- then drop the redundant sentence later "In this example we want to update again ..."

"of the `Function` that are" -> "of a `Function` that are"